### PR TITLE
DSD-1730: Updating to DS 3.0.0 and React 18

### PR DESCRIPTION
**Pull request by 7emansell**
Fixes [JIRA ticket DSD-1730](https://jira.nypl.org/browse/DSD-1730).

## This PR does the following:
- Updates to Design System v3
- Updates to React v18.2
- Updates to Node v18
- Minor changes to maintain header style (in `HeaderLowerNav`)
- Adds new `useMediaQuery` hook for header component, to replace Chakra's
- Corrects types, mostly in `encoreCatalogLogOutTimer`

## How has this been tested?

Locally

### Checklist:

- [x] I have updated the Readme documentation accordingly.
- [x] All new and existing tests passed.

### Front End Review:

- [x] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -34,97 +30,120 @@ export interface HeaderProps {
   isProduction?: boolean;
 }
 
+/**
+ * The useMediaQuery hook returns an array of booleans,
+ * indicating whether the given query matches or queries match.
+ * */
+function useMediaQuery(queryList) {
```
Comment (7emansell) [inline src/components/Header/Header.tsx]: Pull this out into `headerUtils`? It is not used anywhere else

Comment (EdwinGuzman) [reply src/components/Header/Header.tsx]: Yes, please. I missed this in the v3 release so it should be exported by the DS so you and other teams don't have to manually copy the function into a utils file.

File: jest.config.js [hunk 2]
```diff
@@ -1,30 +1,30 @@
 // jest.config.js
 module.exports = {
   collectCoverageFrom: [
-    "**/*.{js,jsx,ts,tsx}",
-    "!**/*.d.ts",
-    "!**/node_modules/**",
+    '**/*.{js,jsx,ts,tsx}',
```
Comment (EdwinGuzman) [inline jest.config.js:4]: Was this an auto-update from prettier?

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -23,87 +24,85 @@ interface FooterProps {
  * as navigational NYPL.org links, social media links, copyright, and
  * NYPL building facade image.
  */
-export const Footer = chakra(
```
Comment (EdwinGuzman) [inline src/components/Footer/Footer.tsx:26]: Darn, is `chakra` another export we need? It seems to work without this wrapper but I'll include it a release soon.

Comment (EdwinGuzman) [reply src/components/Footer/Footer.tsx:26]: Ticket: https://jira.nypl.org/browse/DSD-1735

**Review by EdwinGuzman — COMMENTED** [review]
Add this update to the changelog.

File: jest.config.js (hunk 2 above)
Comment (7emansell) [reply jest.config.js:4]: Yes

File: src/components/Footer/Footer.tsx (hunk 3 above)
Comment (7emansell) [reply src/components/Footer/Footer.tsx:26]: I was confused by this but everything seems to work the same without it (unlike useMediaQuery)

File: jest.config.js (hunk 2 above)
Comment (EdwinGuzman) [reply jest.config.js:4]: :( I prefer double quotes but why did this happen if the configs were not updated? Also, in `src/__tests__/setup.ts` some strings are double quoted but the new addition is single quoted...

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,47 +1,54 @@
...
+- Updates React to v18
+
+## 10/30/23
 
 ### Updates
 
 - Updates the DS version to 2.1.1.
 
-## 10/5
+## 10/5/23
```
Comment (EdwinGuzman) [inline CHANGELOG.md:20]: Good catch.

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Thanks. This is working alright. I'm slightly concerned about how the updated cookies are being set/read but we have work soon to update this feature so it's not a big concern.

Part of this work is also to update to Node 18. Can you update it in the `Dockerfile` and `.travis.yml` file?

File: jest.config.js (hunk 2 above)
Comment (EdwinGuzman) [reply jest.config.js:4]: Is issue is just a personal annoyance. When I save the file locally, my local editor's linting rules updates this to double quotes bu then eslint complains. I'll create another ticket for this, it's not a blocker for this release.

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**